x/incentives/types/proposal_test.go

```diff
@@ -0,0 +1,272 @@
+	suite.Require().Equal("RegisterIncentive", (&RegisterIncentiveProposal{}).ProposalType())
+	suite.Require().Equal("incentives", (&CancelIncentiveProposal{}).ProposalRoute())
+	suite.Require().Equal("CancelIncentive", (&CancelIncentiveProposal{}).ProposalType())
+}
+
+func (suite *ProposalTestSuite) TestRegisterIncentiveProposal() {
+	testCases := []struct {
+		name        string
+		title       string
+		description string
+		incentive   Incentive
+		expectPass  bool
+	}{
+		{
+			"Register incentive - valid",
+			"test",
+			"test desc",
+			Incentive{
+				tests.GenerateAddress().String(),
+				sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(5, 2))},
+				10,
+				time.Now(),
+			},
+			true,
+		},
+		{
+			"Register incentive - invalid missing title ",
+			"",
+			"test desc",
+			Incentive{
+				tests.GenerateAddress().String(),
+				sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(5, 2))},
+				10,
+				time.Now(),
+			},
+			false,
+		},
+		{
+			"Register incentive - invalid missing description ",
+			"test",
+			"",
+			Incentive{
+				tests.GenerateAddress().String(),
+				sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(5, 2))},
+				10,
+				time.Now(),
+			},
+			false,
+		},
+		// Invalid address
+		{
+			"Register incentive - invalid address (no hex)",
+			"test",
+			"test desc",
+			Incentive{
+				"0x5dCA2483280D9727c80b5518faC4556617fb19ZZ",
+				sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(5, 2))},
+				10,
+				time.Now(),
+			},
+			false,
+		},
+		{
+			"Register incentive - invalid address (invalid length 1)",
+			"test",
+			"test desc",
+			Incentive{
+				"0x5dCA2483280D9727c80b5518faC4556617fb19",
+				sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(5, 2))},
+				10,
+				time.Now(),
+			},
+			false,
+		},
+		{
+			"Register incentive - invalid address (invalid length 2)",
+			"test",
+			"test desc",
+			Incentive{
+				"0x5dCA2483280D9727c80b5518faC4556617fb194FFF",
+				sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(5, 2))},
+				10,
+				time.Now(),
+			},
+			false,
+		},
+		{
+			"Register incentive - invalid allocation amount >100% ",
+			"test",
+			"test desc",
+			Incentive{
+				tests.GenerateAddress().String(),
+				sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(101, 2))},
+				10,
+				time.Now(),
+			},
+			false,
+		},
+		{
+			"Register incentive - invalid allocation amount 0%",
+			"test",
+			"test desc",
+			Incentive{
+				tests.GenerateAddress().String(),
+				sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(0, 2))},
+				10,
+				time.Now(),
+			},
+			false,
+		},
+		{
+			"Register incentive - zero epochs",
+			"test",
+			"test desc",
+			Incentive{
+				tests.GenerateAddress().String(),
+				sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(5, 2))},
+				0,
+				time.Now(),
+			},
+			false,
+		},
+		// TODO how to test without panicking?
```

Comment:
@fedekunze how do I test validations from the sdk? In this case its `sdk.ValidateDenom`

Reply:
for testing invalid denoms you can just pass an empty or blank string